### PR TITLE
chore(query): remove udf parser in ast crate

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #issue

### CODE REVIEW

**Comment by sundy-li** [comment]
How about moving `ast` crate into `common/ast` ?

**Comment by Xuanwo** [comment]
> How about moving `ast` crate into `common/ast` ?

I believe ast should become `databend-parser`.

File: src/common/legacy-parser/src/analyzer/analyzer_expr_sync.rs [hunk 1]
```diff
@@ -13,27 +13,25 @@
...
-use common_ast::udfs::UDFTransformer;
 use common_datavalues::prelude::*;
 use common_datavalues::type_coercion::merge_types;
 use common_exception::ErrorCode;
 use common_exception::Result;
 use common_functions::aggregates::AggregateFunctionFactory;
-use common_functions::is_builtin_function;
-use common_meta_types::UserDefinedFunction;
 use common_planners::Expression;
+use sqlparser::ast::BinaryOperator;
```
Comment (Xuanwo) [inline src/common/legacy-parser/src/analyzer/analyzer_expr_sync.rs:22]: cc @andylokandy, maybe we should also implement udf in the new plnner?

**Comment by PsiACE** [comment]
> How about moving ast crate into common/ast ?

We will move all the parts that belong to the query into the query directory, so there is no need to move ast back to common. #7050

File: src/common/legacy-parser/src/analyzer/analyzer_expr_sync.rs (hunk 1 above)
Comment (sundy-li) [reply src/common/legacy-parser/src/analyzer/analyzer_expr_sync.rs:22]: It's already done.

**Review by Xuanwo — APPROVED** [review]

**Review by PsiACE — APPROVED** [review]

**Pull request merged**